node/state/opts.go

```diff
@@ -0,0 +1,10 @@
+package state
+
+import "github.com/celestiaorg/celestia-node/node/node"
+
+// WithKeyringAccName sets the `KeyringAccName` field in the key config.
+func WithKeyringAccName(name string) node.Option {
+	return func(sets *node.Settings) {
+		sets.Cfg.Key.KeyringAccName = name
```

Comment:
The state should have its own `Config` struct that global Config just embeds

Reply:
Ok, so essentially, this Config inside `node/key`	can be moved into `node/state`

Reply:
(answering potential argument to keep `node/key` separate and not the part of `node/state`: lets not do this unless we have a bunch of key related components, we can always modularize a module)

Reply:
The solution is not this directly easy, since it causes a cyclical dependency. But I do like that pattern more as well